src/Microsoft.ML.Data/Evaluators/MulticlassClassificationEvaluator.cs

```diff
@@ -909,8 +955,16 @@ private protected override IDataView CombineOverallMetricsCore(IDataView[] metri
             for (int i = 0; i < metrics.Length; i++)
             {
                 var idv = metrics[i];
+
+                // Change the name of the Top-k-accuracies collection column & remove redundant old TopK output
+                if (_outputTopKAcc != null)
+                {
+                    idv = ChangeAllTopKAccColumnName(idv);
+                    idv = DropColumn(idv, MulticlassClassificationEvaluator.TopKAccuracy);
```

Comment:
Assuming the TopKAccuracy column is visible to users, it would be a breaking change to no longer produce it.

Reply:
`TopKAccuracy` on MulticlassClassification Metrics was and still is visible to the users; what changed is that it's now using the `TopKAccuracyForAllK` to retrieve its value. So this is not a breaking change.

The specific lines of code you've highlighted are for MAML, and the only thing they're doing is renaming the columns related to top k accuracy.

Reply:
Ok. Shouldn't be anything to worry about then, being exposed in only MAML and not the estimator API.

I was concerned about IDataView produced during metric calculation, not just the more visible `MulticlassClassificationMetrics` class. I haven't tried reading the metrics from the IDataView directly, and I'm uncertain if the IDataView is even exposed to the user or just used internally. If the IDataView is exposed to users, in theory, a user could be accessing it. Though, as you said, the code here is only called on a MAML code path.